AK/BigInts.h

```diff
@@ -0,0 +1,486 @@
+    {
+        R res { 0 };
+        div_mod(other, res);
+        return res;
+    }
+
+    template<typename U>
+    constexpr R& operator*=(const U& other)
+    {
+        *this = *this * other;
+        return *this;
+    }
+    template<typename U>
+    constexpr R& operator/=(const U& other)
+    {
+        *this = *this * other;
+        return *this;
+    }
+    template<typename U>
+    constexpr R& operator%=(const U& other)
+    {
+        *this = *this * other;
+        return *this;
+    }
+
+private:
+    T m_low { 0 };
+    T m_high { 0 };
+};
+
+template<typename T>
+struct Formatter<UBigInt<T>> : StandardFormatter {
+    Formatter() = default;
+    explicit Formatter(StandardFormatter formatter)
+        : StandardFormatter(formatter)
+    {
+    }
+
+    void format(FormatBuilder& builder, UBigInt<T> value)
+    {
+        if (!value.high()) {
+            Formatter<T> formatter { *this };
+            return formatter.format(builder, value.low());
+        }
+
+        if (m_precision.has_value())
+            VERIFY_NOT_REACHED();
+
+        if (m_mode == Mode::Pointer) {
+            // this is way to big for a pointer
+            VERIFY_NOT_REACHED();
+        }
+
+        u8 base = 0;
+        bool upper_case = false;
+        if (m_mode == Mode::Binary) {
+            base = 2;
+        } else if (m_mode == Mode::BinaryUppercase) {
+            base = 2;
+            upper_case = true;
+        } else if (m_mode == Mode::Octal) {
+            base = 8;
+        } else if (m_mode == Mode::Decimal || m_mode == Mode::Default) {
+            base = 10;
+        } else if (m_mode == Mode::Hexadecimal) {
+            base = 16;
+        } else if (m_mode == Mode::HexadecimalUppercase) {
+            base = 16;
+            upper_case = true;
+        } else {
+            VERIFY_NOT_REACHED();
+        }
+
+        size_t width = m_width.value_or(0);
+        size_t lower_length = ceil_div(sizeof(T) * 8, base);
+        if (value.high()) {
+            Formatter<T> formatter { *this };
+
+            formatter.m_width = max(width - lower_length, size_t(0u));
+            formatter.format(builder, value.high());
+
+            formatter.m_zeropad = true;
+            formatter.m_alternative_form = false;
+            formatter.m_width = lower_length;
+            formatter.format(builder, value.low());
+        } else {
+            Formatter<T> formatter { *this };
+            formatter.format(builder, value.low());
+        }
+    }
+};
+}
+
+// Nit: Doing these as custom classes might be faster, especially when writing
+//      them in SSE, but this would cause a lot of Code duplication and due to
+//      the nature of constexprs and the intelligence of the compiler they might
+//      be using SSE/MMX either way
+using u128 = AK::UBigInt<u64>;
+using u256 = AK::UBigInt<u128>;
+using u512 = AK::UBigInt<u256>;
+using u1024 = AK::UBigInt<u512>;
+using u2048 = AK::UBigInt<u1024>;
```

Comment:
Possible simplification (or complication):
Draft `AK::UBigInt<size>` where `size` is a power of 2:
```c++
namespace Detail {

template<size_t Size>
requires(IsPowerOfTwo(Size)) // Implement this as some consteval function
struct UBigInt {
    using Type = typename Detail::UBigInt<typename UBigInt<Size/2>::Type>::Type;
};

template<>
struct UBigInt<128> {
     using Type = Detail::UBigInt<u64>;
};
}

template<size_t Size>
using UBigInt = typename Detail::UBigInt<Size>::Type;
```

Not sure if this is useful or even a good idea though tbh

Reply:
well the amount of definitions woudn't really change, due to us having to hardcode every possibility either way, for ease of use

Reply:
Yeah, I guess there's no point in doing this, going past `u2048` or maybe `u4096` is probably crazy anyway :P